lib/common/strings.c

```diff
@@ -41,7 +41,10 @@ crm_itoa_stack(int an_int, char *buffer, size_t len)
  * \param[out] result    If not NULL, where to store scanned value
  * \param[out] end_text  If not NULL, where to store pointer to just after value
  *
- * \return Standard Pacemaker return code (also set errno on error)
+ * \return Standard Pacemaker return code (\c pcmk_rc_ok on success,
+ *         \c EINVAL on failed string conversion due to invalid input,
+ *         or \c ERANGE on arithmetic overflow)
+ * \note Sets \c errno on error
```

Comment:
I'm leaning against this. The strtoll() errno values could be system-dependent, and using strtoll() is an implementation detail that callers shouldn't rely on, so I think the current description is sufficient. It would be fine to call out ERANGE as a particular value of interest. (Even better would be to return EOVERFLOW or pcmk_rc_underflow as appropriate.)

Reply:
I'm hesitant to use pcmk_rc_underflow in this context, since it seems "underflow" usually refers to a floating-point number with a subnormal absolute value. At a minimum, we'd have to use a more generic string for the return code (e.g, "Value underflowed" instead of "Value too small to be stored in data type").

We could define pcmk_rc_int_underflow and pcmk_rc_float_underflow, but even then there's a lack of consistency. A "too negative" int would be considered an underflow, while a "too negative" double would be considered an overflow. Maybe pcmk_rc_neg_underflow and pcmk_rc_zero_underflow or something...

Reply:
I was thinking of negative numbers as smaller than positive numbers, so "too small" could make sense as a message. But as you mentioned, it would be inconsistent with doubles.

I'm thinking mainly in terms of pcmk_rc_str() giving a reasonable message to the user. If we think "value too large" is intuitive enough for a "too negative" double, it also makes sense for "too negative" int.

Reply:
> If we think "value too large" is intuitive enough for a "too negative" double

It might not be ideal, but it is the standard EOVERFLOW message. We'll need to add a special case to pcmk_rc_str() for the EOVERFLOW message or define a pcmk_rc_overflow if we don't like it.

FWIW the full message is "Value too large to be stored in data type", which IMO at least **implies** that the absolute value is the important part.

Reply:
Yeah, I think the situation is less than ideal but acceptable. We can stick with ERANGE or use EOVERFLOW.

Reply:
Switching to EOVERFLOW in next psuh.